src/main/groovy/wooga/gradle/upm/artifactory/traits/UPMPackSpec.groovy

```diff
@@ -64,4 +69,23 @@ trait UPMPackSpec extends BaseSpec {
         generateMetaFiles.set(value)
     }
 
+    private final ListProperty<Consumer<GenerateUpmPackageSpec>> packageJsonPatchers = objects.listProperty(Consumer<GenerateUpmPackageSpec>)
+
+    @Input
+    @Optional
+    ListProperty<Consumer<GenerateUpmPackageSpec>> getPackageJsonPatchers() {
+        return packageJsonPatchers
+    }
+
+    void setPackageJsonPatchers(Provider<? extends Iterable<Consumer<GenerateUpmPackageSpec>>> patchers) {
+        packageJsonPatchers.set(patchers)
+    }
+
+    void setPackageJsonPatchers(Iterable<Consumer<GenerateUpmPackageSpec>> patchers) {
+        packageJsonPatchers.set(patchers)
+    }
+
+    void patchPackageJson(Consumer<GenerateUpmPackageSpec> patcher) {
+        this.packageJsonPatchers.add(patcher)
```

Comment:
I personally feel strongly against the name for this method, as we can be doing more than patching. This is about modifying the packaging task properties. 

I think calling this method `package` would be best, so it goes:

```java
projects {
                defaultPackage {
                    packageDirectory = ${wrap(packageDir.absolutePath, Directory)}
                    package { p ->
                        p.patch("test", "value")
                    }
                }
            }
```